extensions/aerial-photo/tests/aerial_photo_collection.test.js

```diff
@@ -30,6 +30,18 @@ o.spec('Aerial Photo Extension Collection', () => {
     o(valid).equals(true)(JSON.stringify(validate.errors, null, 2));
   });
 
+  o('Missing summaries section should pass validation', async () => {
```

Comment:
Yes, and I thought it would be good to check that it doesn't creep back in at some point. We want to keep these extensions as "light" as possible to make them easier for other people to use.

Reply:
> We want to keep these extensions as "light" as possible to make them easier for other people to use.
Totally agreed! 
My question was focus on the unit test - as this test that a non required field missing is not failing the validation and I did not understand why a test on this field when there are other non required field that don't have a similar test. But I understand your point of view:
> I thought it would be good to check that it doesn't creep back in at some point.

Reply:
Basically it's a regression test.

Reply:
What is you point of view on regression tests @l0b0 ? Should they stay there for good? Should we tag them for the code understanding later?

Reply:
Interesting question. I'll try to answer with some personal "testing logic":

1. The example file should contain examples of *all* the properties defined by the extension. Otherwise, what's the point?
2. The first test shows that the file with all the properties is valid, to make sure we have a sane baseline.
3. Subsequent tests show how this baseline can and  can't be changed. By removing a mandatory property and asserting that the result fails validation we've tested that it is in fact mandatory. By removing an optional property and asserting that the result passes validation we've tested that it is in fact optional.
4. It follows that removing the tests for mandatory *or* optional properties allows bugs to sneak in unnoticed.

So even though this is a regression test, it's not "just" a regression test. It verifies a useful feature of the schema which is not verified by other tests, so it should probably stay there for the foreseeable future.

Reply:
Great answer, thanks @l0b0 !

Reply:
Thanks for the interesting discussion @paulfouquet and @l0b0 , very useful.